scripts/dashboard-node-build.js

```diff
@@ -0,0 +1,206 @@
+#!/usr/bin/env node
+
+/**
+ * Dashboard build script
+ */
+
+const esbuild = require('esbuild')
+const { sassPlugin } = require('esbuild-sass-plugin')
+const path = require('path')
+const fs = require('fs')
+const process = require('process')
+
+const projectRoot = path.dirname(__dirname)
+const dashboardRoot = path.join(projectRoot, 'src', 'serve', 'dashboard')
+const buildDir = path.join(projectRoot, 'dist-dashboard')
+const vueCompiledDir = path.join(projectRoot, '.vue-compiled')
+
+const NODE_ENV = process.env.NODE_ENV || 'development'
+
+console.log('🔧 Building dashboard with Node.js + esbuild...')
+console.log(`📁 Dashboard root: ${dashboardRoot}`)
+console.log(`📦 Build directory: ${buildDir}`)
+console.log(`🌍 Environment: ${NODE_ENV}`)
+
+// Clean and prepare directories
+if (fs.existsSync(buildDir)) {
+  fs.rmSync(buildDir, { recursive: true })
+}
+fs.mkdirSync(path.join(buildDir, 'assets', 'js'), { recursive: true })
+fs.mkdirSync(path.join(buildDir, 'assets', 'css'), { recursive: true })
+fs.mkdirSync(path.join(buildDir, 'assets', 'fonts'), { recursive: true })
+
+// Copy static assets
+function copyStatic() {
+  const assetsDir = path.join(dashboardRoot, 'assets')
+  const targetAssetsDir = path.join(buildDir, 'assets')
+  
+  if (fs.existsSync(assetsDir)) {
+    fs.cpSync(assetsDir, targetAssetsDir, { recursive: true })
+    console.log('✅ Static assets copied')
+  }
+  
+  // Copy index.html
+  const indexHtml = path.join(dashboardRoot, 'index.html')
+  const targetIndex = path.join(buildDir, 'index.html')
+  if (fs.existsSync(indexHtml)) {
+    fs.copyFileSync(indexHtml, targetIndex)
+    console.log('✅ index.html copied')
+  }
+}
+
+copyStatic()
+
+// esbuild configuration
+const buildOptions = {
+  entryPoints: [
+    path.join(vueCompiledDir, 'main.js'),
+    path.join(dashboardRoot, 'assets', 'style', 'main.scss'),
+    path.join(vueCompiledDir, 'combined-styles.scss') // Include extracted Vue component styles
```

Comment:
Created `combined-styles.css` to bundle all Vue component-specific styles (from <style> blocks) into a single file, ensuring all component styles are loaded together instead of being scattered across individual component files, preventing missing styles in the dashboard.